### PR TITLE
refactor: Refactor string input checks

### PR DESCRIPTION
**Pull request by yinggeh**
#### What does the PR do?
Move each individual check of string input in model backend repos to the bakend repo. 

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [x] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [x] refactor

#### Related PRs:
backend: https://github.com/triton-inference-server/backend/pull/101
tensorflow_backend: https://github.com/triton-inference-server/tensorflow_backend/pull/104
pytorch_backend: https://github.com/triton-inference-server/pytorch_backend/pull/136
onnxruntime_backend: https://github.com/triton-inference-server/onnxruntime_backend/pull/263
python_backend: https://github.com/triton-inference-server/python_backend/pull/370

#### Test plan:
- CI Pipeline ID:
17021222 and 17057045

#### Background
Initially just to add missing string input tensor checks to python_backend. Refactored because duplicate code from other model backends were found.

### CODE REVIEW

File: src/onnxruntime.cc [hunk 1]
```diff
@@ -2485,50 +2486,24 @@ ModelInstanceState::SetStringInputBuffer(
...
-                   std::to_string(expected_element_cnt))
-                      .c_str()));
-          break;
-        }
-
-        const uint32_t len = *(reinterpret_cast<const uint32_t*>(data_content));
-        remaining_bytes -= sizeof(uint32_t);
+      TRITONSERVER_Error* err = ValidateStringBuffer(
+          data_content, expected_byte_size, expected_element_cnt,
+          input_name.c_str(), &str_list);
```
Comment (Tabrizian) [inline src/onnxruntime.cc:2492]: If there is an error can we fail all the responses here rather than L2502.

Comment (yinggeh) [reply src/onnxruntime.cc:2492]: Two  `if (err != nullptr)`?
```
      TRITONSERVER_Error* err = ValidateStringBuffer(
          data_content, expected_byte_size, expected_element_cnt,
          input_name.c_str(), &str_list);
      if (err != nullptr) {
        RESPOND_AND_SET_NULL_IF_ERROR(&((*responses)[idx]), err);
      }

      // Set string values.
      for (const auto& [addr, len] : str_list) {
        // Make first byte of size info 0, so that if there is string data
        // in front of it, the data becomes valid C string.
        *const_cast<char*>(addr - sizeof(uint32_t)) = 0;
        string_ptrs->push_back(addr);
      }

      size_t element_cnt = str_list.size();
      if (err != nullptr) {
        FillStringData(string_ptrs, expected_element_cnt - element_cnt);
      }
```

**Review by Tabrizian — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]
